test/common/crs_modifided.go

```diff
@@ -0,0 +1,1973 @@
+		}
+
+		t.Logf("Deleting CR Values : %s: count = %v, revison = %s", cr.Name, count, cr.ResourceVersion)
+		_, err = i.waitReconcilingCR(ctx, cr)
+		if err != nil {
+			t.Fatalf("Deleting CR Values : %s: %s:, %s", cr.Kind, cr.Name, err)
+		}
+		results = i.compareValues(&cr)
+
+		if results == nil {
+			forceRetry = false
+			count -= 1
+		}
+		count -= 1
+		if count < 0 {
+			forceRetry = false
+		}
+	}
+
+	if results != nil {
+		for _, result := range *results {
+			t.Logf("Deleting CR Values : %s: %s: %s: %s", result.Type, result.Name, result.Key, result.Error)
+		}
+		t.Fatal("Deleting CR Values : Failed to reset the CR values")
+	}
+}
+
+func (i *roleBindingCr) addNewValues(t *testing.T, ctx *TestingContext, cr rbacv1.RoleBinding) {
+	err := ctx.Client.Get(goctx.TODO(), k8sclient.ObjectKey{Name: cr.Name, Namespace: cr.Namespace}, &cr)
+	if err != nil {
+		t.Fatal("Add New CR Values :  Failed to refresh CR")
+	}
+	i.addCRValue(cr)
+	err = ctx.Client.Update(goctx.TODO(), &cr)
+	if err != nil {
+		t.Fatal("Add New CR Values :  Failed to update CR on cluster")
+	}
+
+	// Refresh CR to get up-to-date version number
+	err = ctx.Client.Get(goctx.TODO(), k8sclient.ObjectKey{Name: cr.Name, Namespace: cr.Namespace}, &cr)
+	if err != nil {
+		t.Fatalf("Add New CR Values :  Fail to refresh the cr")
+	}
+
+	_, err = i.waitReconcilingCR(ctx, cr)
+	if err != nil && err.Error() != "timed out waiting for the condition" {
+		t.Fatal(err)
+	} else {
+		i.addedValuesStillExist(t, cr)
+	}
+}
+
+func (i *roleBindingCr) copyRequiredValues(cr rbacv1.RoleBinding) {
+	ant := cr.GetAnnotations()
+	i.IntegreatlyName = ant[integreatlyName]
+	i.IntegreatlyNamespace = ant[integreatlyNamespace]
+	i.RoleRefKind = cr.RoleRef.Kind
+	i.RoleRefName = cr.RoleRef.Name
+	for _, subject := range cr.Subjects {
+		i.Subjects = append(i.Subjects, roleBindingSubject{
+			SubjectName: subject.Name,
+			SubjectKind: subject.Kind,
+		})
+	}
+
+}
+
+func (i *roleBindingCr) changeCRValues(cr *rbacv1.RoleBinding) {
+	ant := cr.GetAnnotations()
+	if ant == nil {
+		ant = map[string]string{}
+	}
+	ant[integreatlyName] = "Bad Value"
+	ant[integreatlyNamespace] = "Bad Value"
+	cr.SetAnnotations(ant)
+	//TODO Find a Role Kind that is allowed and is not Kind: Role
+	//cr.RoleRef.Kind = "Bad Value"
+	// Can not change role reference
+	//cr.RoleRef.Name = "bad-value"
+	for index := range cr.Subjects {
+		cr.Subjects[index].Name = "bad-value"
+		cr.Subjects[index].Kind = "ServiceAccount"
+		cr.Subjects[index].APIGroup = ""
+	}
+}
+
+func (i *roleBindingCr) waitReconcilingCR(ctx *TestingContext, cr rbacv1.RoleBinding) (done bool, err error) {
+	resourceVersion := cr.ResourceVersion
+	err = wait.Poll(crRetryInterval, crTimeout, func() (done bool, err error) {
+		err = ctx.Client.Get(goctx.TODO(), k8sclient.ObjectKey{Name: cr.Name, Namespace: cr.Namespace}, &cr)
+		if err != nil {
+			return false, err
+		}
+
+		if resourceVersion != cr.ResourceVersion {
+			return true, nil
+		} else {
```

Comment:
else not required, just:
`return false, nil`

Reply:
What would be standard practice in this case. Personal I like having the else, its make the next step much clearer to me.

Reply:
It's not a big deal either way IMO so I'll leave it up to you. AFAIK, our code linting standards are basically the default from golint, so if its not highlighted by that, then assume it's fine. However, it is slightly superfluous code.